### PR TITLE
FC+elementwise_add (residual connection)

### PR DESCRIPTION
**Pull request by Silv3S**
### PR types
Performance optimization

### PR changes
OPs

### Describe
- Fuse FC with elementwise_add so residual data is added inside FC operator,
- add unit tests with multiple cases for this fuse pass,
- download VIT-OCR model and perform accuracy test.

### Benchmarks
Test were conducted with oneDNN 2.5.4 , `fc_mkldnn_pass` and `fc_act_mkldnn_pass` were enabled:
- VIT-OCR is performing 4% faster (12 fc+residual fusions),
- BERT is performing 8% faster (24 fc+residual fusions).

### CODE REVIEW

**Review by zhiqiu — APPROVED** [review]
LGTM

**Comment by Silv3S** [comment]
Hi @XieYunshen @chalsliu could you please approve setting TIMEOUT for newly added unit test?

**Review by XieYunshen — APPROVED** [review]
LGTM for `set_tests_properties(test_mkldnn_fc_elementwise_add_fuse_pass PROPERTIES TIMEOUT 120)`

**Comment by lidanqing-vv** [comment]
#40834 has been reviewed by many developers and performance and model UT was given. This PR is the same as #40834. Hence I merge it.

**Comment by baoachun** [comment]
Hi @Silv3S, this pr will cause the `test_analyzer_vit_ocr` single test execution of the Windows task to fail, could you plsease check it? https://xly.bce.baidu.com/paddlepaddle/paddle/newipipe/detail/5400245/job/14068509

File: paddle/fluid/inference/tests/api/CMakeLists.txt [hunk 1]
```diff
@@ -344,6 +344,19 @@ inference_analysis_test(test_analyzer_transformer_profile SRCS analyzer_transfor
...
+set(VIT_OCR_INSTALL_DIR "${INFERENCE_DEMO_INSTALL_DIR}/vit_ocr")
+if (NOT EXISTS ${VIT_OCR_INSTALL_DIR}/vit_ocr.tgz)
+    inference_download_and_uncompress_without_verify(${VIT_OCR_INSTALL_DIR} ${VIT_OCR_URL} vit_ocr.tgz)
+endif()
+if (NOT EXISTS ${VIT_OCR_INSTALL_DIR}/datavit.txt)
+    file(DOWNLOAD ${VIT_OCR_URL}/datavit.txt ${VIT_OCR_INSTALL_DIR}/datavit.txt)
+endif()
+inference_analysis_test(test_analyzer_vit_ocr SRCS analyzer_vit_ocr_tester.cc
+  EXTRA_DEPS ${INFERENCE_EXTRA_DEPS}
+  ARGS --infer_model=${VIT_OCR_INSTALL_DIR}/vit_ocr --infer_data=${VIT_OCR_INSTALL_DIR}/datavit.txt)
```
Comment (lidanqing-vv) [inline paddle/fluid/inference/tests/api/CMakeLists.txt:358]: @Silv3S Hi, this new test failed on windows, please investigate thank you.

**Comment by Silv3S** [comment]
@baoachun @lidanqing-intel  Thank you for letting me know. As I can see there is error with downloading model and data for test. That's unexpected because all windows tests were ok on PR CI and `test_analyzer_vit_ocr` works locally for me. I will investigate more and fix it

**Comment by lidanqing-vv** [comment]
Hi, @baoachun . Siwek checked it seems models downloading issue and the link address is rarely used in API tests before. Please move vit_ocr.tgz and datavit.txt to more frequently used link and combine them into one tgz.

original model link
https://paddle-qa.bj.bcebos.com/inference_model/2.1.1/ocr/vit_ocr.tgz
original data link
https://paddle-qa.bj.bcebos.com/inference_model/2.1.1/ocr/datavit.txt

Move the two files into one http://paddle-inference-dist.bj.bcebos.com/ocr/vit_ocr.tgz. 
Its distracted files should look like
```
- vit_ocr/
  - datavit.txt
  - model/
    - inference.pdiparams
    - inference.pdmodel  
```
Thanks.

**Comment by Silv3S** [comment]
I changed config, and now VIT-OCR model and data will be downloaded from same source as other models https://github.com/PaddlePaddle/Paddle/pull/42041

**Pull request merged**